src/transforms/centroid.js

```diff
@@ -28,6 +29,55 @@ export function centroid({geometry = identity, ...options} = {}) {
   );
 }
 
+export function poi({geometry = identity, ...options} = {}) {
+  const getG = memoize1((data) => valueof(data, geometry));
+  return initializer(
+    {...options, x: null, y: null, geometry: {transform: getG}},
+    (data, facets, channels, scales, dimensions, {projection}) => {
+      const G = getG(data);
+      const n = G.length;
+      const X = new Float64Array(n);
+      const Y = new Float64Array(n);
+      let polygons, holes, ring;
+      const alpha = 2;
+      const context = {
+        arc() {},
+        moveTo(x, y) {
+          ring = [[x, -alpha * y]];
+        },
+        lineTo(x, y) {
+          ring.push([x, -alpha * y]);
+        },
+        closePath() {
+          ring.push(ring[0]);
+          if (polygonArea(ring) > 0) polygons.push([ring]);
+          else holes.push(ring);
+        }
+      };
+      const path = geoPath(projection, context);
```

Comment:
Tested to work with #2252:

```suggestion
      const path = context.path().context({
        arc() {},
        moveTo(x, y) {
          ring = [[x, -alpha * y]];
        },
        lineTo(x, y) {
          ring.push([x, -alpha * y]);
        },
        closePath() {
          ring.push(ring[0]);
          if (polygonArea(ring) > 0) polygons.push([ring]);
          else holes.push(ring);
        }
      });
```